bll/Exceptions/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Data.SqlClient;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace bll.Exceptions
+{
+    public class ExceptionHandlingMiddleware
```

Comment:
This kinda works of course... But better take a look at ExceptionFilterAttribute and what you can do with inheriting it in a custom attribute that you can apply to all controllers with something like this is startup class: 
```
services.AddControllers(options =>
            {
                options.Filters.Add<ApiExceptionFilterAttribute>();
            })
```
Here we assume ApiExceptionFilterAttribute is inherited from Microsoft's ExceptionFilterAttribute and looks similar to this:
```
public class ApiExceptionFilterAttribute : ExceptionFilterAttribute
    {
        private readonly IDictionary<Type, Action<ExceptionContext>> exceptionHandlers;

        public ApiExceptionFilterAttribute()
        {
            this.exceptionHandlers = new Dictionary<Type, Action<ExceptionContext>>
            {
                { typeof(NotFoundException), this.HandleNotFoundException }
            };
        }

        public override void OnException(ExceptionContext context)
        {
            this.HandleException(context);
            base.OnException(context);
        }

        private void HandleException(ExceptionContext context)
        {
            Type type = context.Exception.InnerException == null ?
                context.Exception.GetType() :
                context.Exception.InnerException.GetType();
            
            if (this.exceptionHandlers.ContainsKey(type))
            {
                this.exceptionHandlers[type].Invoke(context);
                return;
            }

            HandleUnknownException(context);
        }

        private void HandleNotFoundException(ExceptionContext context)
        {
            var exception = context.Exception as NotFoundException ??
                            context.Exception.InnerException as NotFoundException;

            context.Result = new NotFoundObjectResult(new NotFoundResponse(){Message = exception.Message});

            context.ExceptionHandled = true;
        }
    }
```

Reply:
Of course in your app your idea will work fine as long as you don't have anything other than simple REST Api. But ExceptionFilterAttribute approach helps us to avoid overriding exceptions in background services, signalR handlers and all other not controller-related places. It also looks better in my opinion.